### PR TITLE
jasmine: update Env

### PR DESCRIPTION
**Pull request by UziTech**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/jasmine/jasmine/blob/main/src/core/Env.js>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

---

Update Jasmine.Env and config options

### CODE REVIEW

**Comment by UziTech** [comment]
TODO:
- [ ] Rebase on #51997 after it is merged
- [ ] Add tests

**Comment by chivesrs** [comment]
Just make sure you're not duplicating work that's being done by one of the Jasmine core maintainers in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/51998

File: types/jasmine/index.d.ts [hunk 1]
```diff
@@ -435,59 +482,93 @@ declare namespace jasmine {
     }
 
     interface Env {
+        addAsyncMatchers: typeof addAsyncMatchers;
```
Comment (chivesrs) [inline types/jasmine/index.d.ts:485]: This value isn't documented at https://jasmine.github.io/api/edge/Env

It seems to not be part of the public API. Is this intentional?

**Review by chivesrs — CHANGES_REQUESTED** [review]

File: types/jasmine/index.d.ts (hunk 1 above)
Comment (UziTech) [reply types/jasmine/index.d.ts:485]: None of the global or static function like [`addAsyncMatchers`](https://jasmine.github.io/api/edge/jasmine.html#.addAsyncMatchers) are documented on `Env` but some of them were already in the types so I wasn't sure If I should add all of them.

All of those functions are public.

Comment (chivesrs) [reply types/jasmine/index.d.ts:485]: I would err on the side of not adding them, unless the project you are working on needs them. But, I'll approve this and let you decide.

@sgravrock Any chance you can comment on this, if you see this?

**Review by chivesrs — APPROVED** [review]

**Comment by sgravrock** [comment]
Yeah, methods like Env#addAsyncMatchers aren’t really intended to be public. If there’s a reason why they need to be used from outside Jasmine we can talk about that, but otherwise I think it would be better not to add them.

**Comment by UziTech** [comment]
I removed the global and jasmine static functions from Env. It doesn't look like any other types relied on them.

File: types/jasmine/index.d.ts [hunk 2]
```diff
@@ -436,58 +483,66 @@ declare namespace jasmine {
...
-        execute(): void;
-        describe(description: string, specDefinitions: () => void): Suite;
-        // ddescribe(description: string, specDefinitions: () => void): Suite; Not a part of jasmine. Angular team adds these
-        beforeEach(beforeEachFunction: ImplementationCallback, timeout?: number): void;
-        beforeAll(beforeAllFunction: ImplementationCallback, timeout?: number): void;
-        afterEach(afterEachFunction: ImplementationCallback, timeout?: number): void;
-        afterAll(afterAllFunction: ImplementationCallback, timeout?: number): void;
-        xdescribe(desc: string, specDefinitions: () => void): XSuite;
-        it(description: string, func: () => void): Spec;
+        allowRespy(allow: boolean): void;
```
Comment (chivesrs) [inline types/jasmine/index.d.ts:486]: I do not see this in https://jasmine.github.io/api/3.7/Env

File: types/jasmine/index.d.ts [hunk 3]
```diff
@@ -436,58 +483,66 @@ declare namespace jasmine {
...
+        clearReporters(): void;
+        configuration(): EnvConfiguration;
+        configure(configuration: EnvConfiguration): void;
+        execute(runnablesToRun?: Suite[], onComplete?: Func): void;
```
Comment (chivesrs) [inline types/jasmine/index.d.ts:490]: Same

File: types/jasmine/index.d.ts (hunk 2 above)
Comment (chivesrs) [reply types/jasmine/index.d.ts:486]: Can you mark this as `@deprecated`? It seems like it's a preexisting type that should no longer exist.

File: types/jasmine/index.d.ts [hunk 4]
```diff
@@ -436,58 +483,66 @@ declare namespace jasmine {
...
+        /**
+         * @deprecated Use hideDisabled option in {@link jasmine.Env.configure} instead.
+         */
+        hideDisabled(value: boolean): void;
+        /**
+         * @deprecated Check hideDisabled option in {@link jasmine.Env.configuration} instead.
+         */
+        hidingDisabled(): boolean;
```
Comment (chivesrs) [inline types/jasmine/index.d.ts:498]: This isn't in the docs right now nor was it in the old types. Remove?

File: types/jasmine/index.d.ts [hunk 5]
```diff
@@ -436,58 +483,66 @@ declare namespace jasmine {
...
+        provideFallbackReporter(reporter: CustomReporter): void;
         /**
-         * @deprecated Use oneFailurePerSpec option in {@link jasmine.Env.configure} instead.
+         * @deprecated Use random option in {@link jasmine.Env.configure} instead.
          */
-        throwOnExpectationFailure(value: boolean): void;
+        randomTests(): boolean;
         /**
-         * @deprecated Use failFast option in {@link jasmine.Env.configure} instead.
+         * @deprecated Check random option in {@link jasmine.Env.configuration} instead.
```
Comment (chivesrs) [inline types/jasmine/index.d.ts:506]: Use, not check

This is a setter not a getter.

Comment (chivesrs) [reply types/jasmine/index.d.ts:506]: I believe the JSDoc should be identical to L502?

File: types/jasmine/index.d.ts [hunk 6]
```diff
@@ -436,58 +483,66 @@ declare namespace jasmine {
...
-        configure(configuration: EnvConfiguration): void;
+        specFilter(spec: Spec): boolean;
+        /**
+         * @deprecated Use failFast option in {@link jasmine.Env.configure} instead.
+         */
+        stopOnSpecFailure(value: boolean): void;
+        /**
+         * @deprecated Check failFast option in {@link jasmine.Env.configuration} instead.
+         */
+        stoppingOnSpecFailure(): boolean;
```
Comment (chivesrs) [inline types/jasmine/index.d.ts:536]: Remove?

File: types/jasmine/index.d.ts [hunk 7]
```diff
@@ -436,58 +483,66 @@ declare namespace jasmine {
...
+        /**
+         * @deprecated Use oneFailurePerSpec option in {@link jasmine.Env.configure} instead.
+         */
+        throwOnExpectationFailure(value: boolean): void;
+        /**
+         * @deprecated Check oneFailurePerSpec option in {@link jasmine.Env.configuration} instead.
+         */
+        throwingExpectationFailures(): boolean;
+        topSuite(): Suite;
```
Comment (chivesrs) [inline types/jasmine/index.d.ts:545]: This isn't in the docs either.

**Review by chivesrs — CHANGES_REQUESTED** [review]

File: types/jasmine/index.d.ts (hunk 2 above)
Comment (UziTech) [reply types/jasmine/index.d.ts:486]: https://github.com/jasmine/jasmine/commit/2fc5182ddcdd67456b89c92c2eedbc2443c928c6#diff-46cdc2a30e1879f73f5a62a709840911600e79ff44702785f1fafb4234789c57R1919

This is not deprecated

File: types/jasmine/index.d.ts (hunk 3 above)
Comment (UziTech) [reply types/jasmine/index.d.ts:490]: same

File: types/jasmine/index.d.ts (hunk 4 above)
Comment (UziTech) [reply types/jasmine/index.d.ts:498]: This is a deprecated method. I figured it was best to include it to make the types more accurate.

File: types/jasmine/index.d.ts (hunk 7 above)
Comment (UziTech) [reply types/jasmine/index.d.ts:545]: https://github.com/jasmine/jasmine/commit/ce850c472a69b59b10abb24d93b6955e864b078a#diff-46cdc2a30e1879f73f5a62a709840911600e79ff44702785f1fafb4234789c57R1628

same

File: types/jasmine/index.d.ts (hunk 5 above)
Comment (UziTech) [reply types/jasmine/index.d.ts]: they were switched

File: types/jasmine/index.d.ts (hunk 7 above)
Comment (UziTech) [reply types/jasmine/index.d.ts:545]: The docs were added recently so they are in [edge](https://jasmine.github.io/api/edge/Env.html)

**Review by chivesrs — APPROVED** [review]

**Review by chivesrs — APPROVED** [review]

**Comment by UziTech** [comment]
Ready to merge

**Pull request merged**